arithmetization/src/main/java/net/consensys/linea/extradata/LineaExtraDataHandler.java

```diff
@@ -25,89 +25,73 @@
 import org.apache.tuweni.units.bigints.UInt32;
 import org.hyperledger.besu.datatypes.Wei;
 import org.hyperledger.besu.datatypes.rpc.JsonRpcResponseType;
-import org.hyperledger.besu.plugin.data.AddedBlockContext;
-import org.hyperledger.besu.plugin.data.BlockHeader;
-import org.hyperledger.besu.plugin.services.BesuEvents;
-import org.hyperledger.besu.plugin.services.BlockchainService;
 import org.hyperledger.besu.plugin.services.RpcEndpointService;
 
 @Slf4j
-public class LineaExtraDataHandler implements BesuEvents.BlockAddedListener {
+public class LineaExtraDataHandler {
   private final RpcEndpointService rpcEndpointService;
-  private final ExtraDataParser[] extraDataParsers;
+  private final ExtraDataConsumer[] extraDataConsumers;
 
   public LineaExtraDataHandler(
       final RpcEndpointService rpcEndpointService,
-      final BlockchainService blockchainService,
       final LineaProfitabilityConfiguration profitabilityConf) {
     this.rpcEndpointService = rpcEndpointService;
-    extraDataParsers = new ExtraDataParser[] {new Version1Parser(profitabilityConf)};
-    onStartup(blockchainService);
+    extraDataConsumers = new ExtraDataConsumer[] {new Version1Consumer(profitabilityConf)};
   }
 
-  private void onStartup(final BlockchainService blockchainService) {
-    consumeExtraData(blockchainService.getChainHeadHeader());
-  }
-
-  @Override
-  public void onBlockAdded(final AddedBlockContext addedBlockContext) {
-    consumeExtraData(addedBlockContext.getBlockHeader());
-  }
-
-  private void consumeExtraData(final BlockHeader blockHeader) {
-    final var rawExtraData = blockHeader.getExtraData();
+  public void handle(final Bytes rawExtraData) throws LineaExtraDataException {
 
     if (!Bytes.EMPTY.equals(rawExtraData)) {
-      for (final ExtraDataParser extraDataParser : extraDataParsers) {
-        if (extraDataParser.canParse(rawExtraData)) {
+      for (final ExtraDataConsumer extraDataConsumer : extraDataConsumers) {
+        if (extraDataConsumer.canConsume(rawExtraData)) {
           final var extraData = rawExtraData.slice(1);
```

Comment:
could you explain why we need to do this slice ?

Reply:
added a comment